packages/http-client-csharp/generator/Microsoft.Generator.CSharp.ClientModel/src/ScmTypeFactory.cs

```diff
@@ -23,6 +24,46 @@ public class ScmTypeFactory : TypeFactory
 
         public virtual CSharpType TokenCredentialType => throw new NotImplementedException("Token credential is not supported in Scm libraries yet");
 
+        public virtual CSharpType ClientResponseType => typeof(ClientResult);
+
+        public virtual CSharpType ClientResponseOfTType => typeof(ClientResult<>);
+
+        public virtual CSharpType ClientResponseExceptionType => typeof(ClientResultException);
```

Comment:
Per review with @m-nash, should these go in ClientResponseApi?

Reply:
updated